scopes/workspace/workspace/workspace-component/workspace-component-loader.ts

```diff
@@ -272,7 +272,7 @@ export class WorkspaceComponentLoader {
 
   private async upsertExtensionData(component: Component, extension: string, data: any) {
     const existingExtension = component.state.config.extensions.findExtension(extension);
-    if (existingExtension) {
+    if (existingExtension && data) {
```

Comment:
didn't this allow removing data by setting undefined?

Reply:
no. you should never use this to remove data. by design